src/pubsub-emulator.js

```diff
@@ -291,6 +289,8 @@ class PubSubEmulator{
     if (!this._options.topics || this._options.topics.length < 1)
       return Promise.resolve();
 
+    const PubSub = require('@google-cloud/pubsub');
```

Comment:
Why did u move the `this._pubsub` initialization here? I don't remember why I din't import `require('@google-cloud/pubsub')` in the top of the file.

It could break the `isTopicExists` if a topic does not exist in constructor options.

The prev version was wrong too. :)

We have 3 options:
- init `this._pubsub` in the constructor
- create an `initPubSub` method that checks `this._pubsub` inited or not if inited do nothing if not init it.
- create a function that return a new PubSub instance and we call it before create or check the  topic. Finally we remove the `this._pubsub` object variable.

I prefer the 3 solution.

Reply:
To be honest, I would remove automatic topic creation and thus `@google-cloud/pubsub`dependency. Creating topics automatically gives you only false positive assumptions. You should always create topics / subscriptions with your own options inside your unit tests or assume they are already there.

Reply:
I agree with you, but maybe others use this feature and I don't really want to make trouble to them.